packages/wmr/src/wmr-middleware.js

```diff
@@ -284,7 +284,14 @@ export default function wmrMiddleware(options) {
 		} else if (queryParams.has('asset')) {
 			cacheKey += '?asset';
 			transform = TRANSFORMS.asset;
-		} else if (prefix || hasIdPrefix || isModule || /\.([mc]js|[tj]sx?)$/.test(file) || /\.(css|s[ac]ss)$/.test(file)) {
+		} else if (
+			prefix ||
+			hasIdPrefix ||
+			isModule ||
+			/\.([mc]js|[tj]sx?)$/.test(file) ||
+			/\.(css|s[ac]ss)$/.test(file) ||
+			(extname(file) !== '' && extname(file) !== 'html' && !/favicon\.ico$/.test(file))
```

Comment:
I'm thinking maybe it would make sense to only intercept imports here? `import './foo.md'` is transformed to `import './foo.md?module'` and can be intercepted by plugins, but `fetch('/foo.md')` remains a simple static file.

```suggestion
			(isModule && extname(file) !== '' && extname(file) !== 'html' && !/favicon\.ico$/.test(file))
```

Reply:
The `isModule` check is already present a few lines above, but your snippet gave me an idea. With the recent commits we now append `?module` after transforming the `.js` files, similar to how we handle style imports.